typescript/packages/subsurface-viewer/src/layers/grid3d/webworker.ts

```diff
@@ -839,6 +833,95 @@ export function makeFullMesh(e: { data: WebWorkerParams }) {
         return normalize(res);
     };
 
+    interface IPrimitiveCounts {
+        triangles: number;
+        lineSegments: number;
+    }
+
+    interface IMeshArrays {
+        trianglePoints: Float32Array;
+        triangleNormals: Float32Array;
+        properties: Float32Array;
+        lineIndices: Uint32Array;
+    }
+
+    const getPrimitiveCounts = (polys: Uint32Array): IPrimitiveCounts => {
+        let triangles = 0;
+        let lineSegments = 0;
+        let i = 0;
+        while (i < polys.length) {
+            triangles += polys[i] - 2;
```

Comment:
Why -2 ?

Reply:
3 points - 1 triangle, 4 points - 2 triangles.. etc.

Reply:
OK. The polys are defined as triangle strip.
It's worth a comment...

Reply:
Comments added, docs updated